Add documentation for "Options Menu fails to clear joypad state on initialization" (#984)
pret/pokecrystal: docs/bugs_and_glitches.md

  - [`ReadObjectEvents` overflows into `wObjectMasks`](#readobjectevents-overflows-into-wobjectmasks)
  - [`ClearWRAM` only clears WRAM bank 1](#clearwram-only-clears-wram-bank-1)
  - [`BattleAnimCmd_ClearObjs` only clears the first 6⅔ objects](#battleanimcmd_clearobjs-only-clears-the-first-6-objects)

>   - [Options Menu fails to clear joypad state on initialization](#options-menu-fails-to-clear-joypad-state-on-initialization)

Reviewer: ```suggestion
  - [Options menu fails to clear joypad state on initialization](#options-menu-fails-to-clear-joypad-state-on-initialization)
```

 	jr nz, .loop
 	ret
```

> ### Options Menu fails to clear joypad state on initialization
> 
> This bug (or feature!) results in all options being shifted left or right if the respective direction is pressed on the same frame the options menu is opened. The bug also exists in pokeyellow and pokegold.

Reviewer: ```suggestion
### Options menu fails to clear joypad state on initialization

This bug allows all the options to be updated at once if the left or right buttons are pressed on the same frame that the options menu is opened.
```

Author: Mine was simply a c&p of the wording in the pokeyellow docs. Yours wording is much better. Thanks.

**Fix:** Edit `_Option` in [engine/menus/options_menu.asm](https://github.com/pret/pokecrystal/blob/master/engine/menus/options_menu.asm):

```diff

>  _Option:
> +	call ClearJoypad
> 	ld hl, hInMenu
> 	ld a, [hl]
> 	push af

Reviewer: ```suggestion
  _Option:
+	call ClearJoypad
 	ld hl, hInMenu
 	ld a, [hl]
 	push af
```

Reviewer 2: Gee Rangi, how come your mom lets you have *TWO* spaces in front of `_Option:`?

Reviewer 3: ```suggestion
 _Option:
+	call ClearJoypad
 	ld hl, hInMenu
 	ld a, [hl]
 	push af
```

Reviewer 3: ```suggestion
 _Option:
 -; BUG: Options menu fails to clear joypad state on initialization (see docs/bugs_and_glitches.md)
+	call ClearJoypad
 	ld hl, hInMenu
 	ld a, [hl]
 	push af
```
Likewise that `; BUG:` comment should be `added` to the same location in `engine/menus/options_menu.asm`

 	jr nz, .loop
 	ret
```

### Options menu fails to clear joypad state on initialization

> [Video](https://www.youtube.com/watch?v=uhDSIkXkl3g)

Reviewer: ```suggestion
([Video](https://www.youtube.com/watch?v=uhDSIkXkl3g))
```
